drivers/include/epd_bw_spi.h

```diff
@@ -0,0 +1,251 @@
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    drivers_epd_bw_spi Generic black/white e-paper/e-ink SPI display driver.
+ * @ingroup     drivers_actuators
+ * @brief       Device driver for black/white e-ink/e-paper SPI displays.
+ *
+ * TODO
+ *
+ * @{
+ * @file
+ * @brief       Generic black/white e-paper/e-ink display SPI driver.
+ *
+ * @author      Silke Hofstra <silke@slxh.eu>
+ */
+
+#ifndef EPD_BW_SPI_H
+#define EPD_BW_SPI_H
+
+#include "periph/spi.h"
+#include "periph/gpio.h"
+
+#ifdef MODULE_DISP_DEV
+#include "disp_dev.h"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define EPD_BW_SPI_COLOR_WHITE (0xFF) /**< White (8x1 pixels) */
+#define EPD_BW_SPI_COLOR_BLACK (0x00) /**< Black (8x1 pixels) */
+
+/**
+ * @brief   Data entry mode settings.
+ *
+ * This setting affect the automatic increment/decrement of the address counters.
+ */
+typedef enum {
+    EPD_BW_SPI_Y_DEC_X_DEC = 0b00, /**< Y decrement, X decrement */
+    EPD_BW_SPI_Y_DEC_X_INC = 0b01, /**< Y decrement, X increment */
+    EPD_BW_SPI_Y_INC_X_DEC = 0b10, /**< Y increment, X decrement */
+    EPD_BW_SPI_Y_INC_X_INC = 0b11, /**< Y increment, X increment */
+} epd_bw_spi_entry_mode_t;
+
+/**
+ * @brief   Additional status codes for monochrome SPI epaper displays.
+ */
+enum {
+    EPD_BW_SPI_DC_FAIL = -5,
+    EPD_BW_SPI_RST_FAIL = -6,
+    EPD_BW_SPI_BUSY_FAIL = -7,
+};
+
+/**
+ * @brief   SPI display device initialisation parameters.
+ */
+typedef struct {
+    spi_t spi;          /**< SPI device that the display is connected to */
+    spi_clk_t spi_clk;  /**< SPI clock speed to use */
+    gpio_t cs_pin;      /**< pin connected to the CHIP SELECT line */
+    gpio_t dc_pin;      /**< pin connected to the DC line */
+    gpio_t rst_pin;     /**< pin connected to the reset line */
+    gpio_t busy_pin;    /**< pin connected to the busy line */
+    bool busy_value;    /**< expected value for the busy pin */
+    bool dummy;         /**< if device requires a dummy cycle before read */
+} epd_bw_spi_params_t;
+
+/**
+ * @brief   Display controller parameters.
+ */
+typedef struct {
+    uint8_t vcom;             /**< VCOM voltage level */
+    const uint16_t size_x;    /**< supported number of horizontal pixels */
+    const uint16_t size_y;    /**< supported number of vertical pixels */
+    const uint8_t lut_size;   /**< size of the waveform lookup table */
+    const uint8_t* lut_full;  /**< lookup table for a full display refresh */
+    const uint8_t* lut_part;  /**< lookup table for a partial display refresh */
+} epd_bw_spi_controller_t;
+
+/**
+ * @brief   Device initialisation parameters.
+ */
+typedef struct {
+    #ifdef MODULE_DISP_DEV
+    disp_dev_t *dev;                    /**< Pointer to the generic display device */
+    bool refresh_partial;               /**< Use partial instead of full refreshes */
+    #endif
```

Comment:
```suggestion
#ifdef MODULE_DISP_DEV
    disp_dev_t *dev;                    /**< Pointer to the generic display device */
#endif
```

See my comment above about the partial_refresh attribute.